### ISSUE TITLE
[BUG] [1.16.5] Assistant cook keeps what he makes

### ISSUE THREAD

**Issue by Sandriell** [ldtteam/minecolonies#8103]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest MineColonies Verison?

- [X] I am running the latest **alpha** version of MineColonies for my Minecraft version.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### Minecraft Version

1.16

### MineColonies Version

minecolonies-1.16.5-1.0.462-ALPHA

### Structurize Version

structurize-1.16.5-1.0.343-RELEASE

### Related Mods and their Versions

Complete mod list: https://gist.github.com/Sandriell/0688b4e8e61c46e1bd44c36ad9ff6fc0

### Current Behavior

The assistant cook is very greedy and is not putting what he makes into the inventory of the restaurant and is just keeping it in his own inventory. With a minimum stock set, it results in an endless loop.

I taught the restaurant the recipe for carrot bread from Pam's Harvestcraft and set a minimum stock of one stack. The assistant cook requested all the ingredients, gets them delivered, makes the stack of carrot bread and then just keeps it. He never deposits the finished food into the restaurants inventory so another stack is immediately requested to try and fulfill the minimum stock order.

Tested to see if this was a Harvestcraft issue in particular and it also occurs with beetroot soup.

Additionally, two other related issues I noticed during this is that the assistant cook crafts entire stacks instantly once he has the ingredients. Seems like the should be smacking a crafting table or the hut block for awhile like other crafters do. The second issue is that the assistant cook struggles to recognize that he has the Harvestcraft cooking utensils (such as the bakeware) and still makes a request for them even if they are in his own inventory.

### Expected Behavior

After finishing the request, the food is deposited into the hut's inventory.

### Reproduction Steps

1. Teach the assistant cook a recipe such as beetroot soup.
2. Set a minimum stock for that item.
3. Wait for/supply all the ingredients.
4. Assistant cook makes the request (instantly as pointed out)
5. Immediately requests to make another stock to try and satisfy the minimum stock requirement

### Logs

N/A

### Anything else?

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by MotionlessTrain** [comment]
This doesn't just happen when the restaurant has a minimum stock, but I have seen the same behaviour with golden apples requested by the hospital. The request fails, when the courier can't find the golden apples at the restaurant (the assistant cook still has them in their inventory)

**Comment by MotionlessTrain** [comment]
Assistant cooks seems to get confused when they have the same food type in there inventory as well. After requesting 41 carrots (for different researches), the assistant cook kept five golden carrots in their inventory, and I only got 36 carrots delivered. I requested five more, and the assistant cook only crafted one more (probably thinking he had enough golden carrots already) and put that one golden carrot in the storage. The courier brought just that one golden carrot over. When I requested four golden carrots afterwards, the same happens. And again with three golden carrots

Because of a failed crafting earlier, there are more than enough golden carrots in the restaurant available (and the cook is not allowed to hand those out). So the courier could just get the five (or four or three) golden carrots, even though the cook just (incorrectly) crafted one
